src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -338,12 +118,29 @@ void SyncInferRequest::CopyBlobIfNeeded(InferenceEngine::Blob::CPtr src, Inferen
     }
 }
 
-void SyncInferRequest::CopyOutputsIfNeeded() {
-    for (const auto& it : _networkOutputs) {
-        auto& name = it.first;
+void SyncInferRequest::copy_outputs_if_needed() {
+    for (const auto& it : get_outputs()) {
         // this request is already in BUSY state, so using the internal functions safely
-        CopyBlobIfNeeded(m_batched_request_wrapper._inferRequestBatched->GetBlob(name), GetBlob(name), false);
+        auto dst_tensor = get_tensor(it);
+        copy_tensor_if_needed(m_batched_request_wrapper->_infer_request_batched->get_tensor(it), dst_tensor, false);
     }
 }
+
+void SyncInferRequest::infer() {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::vector<ov::SoPtr<ov::IVariableState>> SyncInferRequest::query_state() const {
+    auto states = m_batched_request_wrapper->_infer_request_batched->query_state();
+    for (auto&& state : states) {
+        if (!state._so)
+            state._so = m_batched_request_wrapper->_infer_request_batched._so;
```

Comment:
are we sure this logic is correct? what if this batch infer request is not been executed with _infer_request_batched?

Reply:
@songbell  you mean when inference task fallback to m_request_without_batch, we shouldn't query state from _infer_request_batched? But fallback should be not normal state, if we still query state from _infer_request_batched it should return the correct current state, right?

Reply:
For `query_state()`, I think we need to follow `get_profiling_info()` overload it in `ov::autobatch_plugin::AsyncInferRequest` class. Will be in next PR.